### PR TITLE
Feat/new category

### LINKED ISSUE #8: #3 create categories
## Value Proposition
As a user, I also want to create categories so that i can sort my list-items inside of them.
## Description
user-story-03
**Category-List Example for localStorage & useState:**
```javascript
[
  {
    id: 'A1259d6j6s',
    name: 'Haushalt'
  },
  {
    id: 'B6ghu86j6s',
    name: 'Haustiere'
  }
]
```

## Acceptance Criteria
- [x] (create-entry page) display input-field to insert a new category
- [x] save the data, so i can reload and get all inserted data
## Tasks
- [x] create feature `feat/new-category`
- [x] add new `Input component` (insert new category) inside the create-entry page
- [x] save the list-items & categories localStorage and useState

### PR DESCRIPTION
**Pull request by philmetscher**
Heyho, i added
- some GlobalStyle fixes
- the ability to create new categories, if you create new list-items (click on "neuer Eintrag")
  - removed the ability to select a category, if you filled the "new category" input

Thanks for your feedback in advance 🙏

### CODE REVIEW

File: components/Category.js [hunk 1]
```diff
@@ -10,12 +10,10 @@ import useLocalStorage from "../hooks/useLocalStorage";
 
 export default function Category({ category }) {
   const [isExtended, setIsExtended] = useState(true);
-  const [listItems, setListItems] = useLocalStorage("listItems", []);
-  useEffect(() => {
-    setListItems(
-      JSON.parse(localStorage.getItem("listItems")) || exampleListItems
```
Comment (Ozanyurdakul98) [inline components/Category.js:16]: Jup, noone likes Json

File: components/FormComponents.js [hunk 2]
```diff
@@ -86,13 +95,14 @@ function Select({
   return (
     <Group>
       <Label htmlFor={name}>{labelText}</Label>
-      <Selection className="noselect">
+      <Selection className="noselect" disabled={disabled}>
         <SelectionHeader
           className="noselect"
-          onClick={toggling}
+          onClick={disabled ? () => setIsOpen(false) : toggling}
```
Comment (Ozanyurdakul98) [inline components/FormComponents.js:110]: can you just use setIsOpen(false) here? I mean getting rid of the Callback thing's

File: pages/create-entry.js [hunk 3]
```diff
@@ -17,33 +17,74 @@ import { useState, useEffect } from "react";
...
+    if (!enterInInput) {
       router.push(`/`);
+    } else {
+      setEnterInInput(false);
     }
   }
   function handleInput(event) {
     const value = event.target.value;
-    value.length > 2 ? setInputTextFilled(true) : setInputTextFilled(false);
+    value.length > 2 ? setSubmitReady(true) : setSubmitReady(false);
```
Comment (Ozanyurdakul98) [inline pages/create-entry.js:111]: Can you short this by using the min and max attribute on the input element? just curios

**Review by Ozanyurdakul98 — COMMENTED** [review]
Just a few comments, hit me back in your slack thread for green

File: components/FormComponents.js [hunk 4]
```diff
@@ -42,9 +43,11 @@ function Input({
         iconBefore={iconBefore}
         placeholder={children}
         onChange={(event) => handleChange(event)}
+        onKeyDown={(event) => handleKeyPress(event)}
```
Comment (anhthu253) [inline components/FormComponents.js:49]: ich glaub man muss kein event as parameter reinpacken. Der kommt automatisch mit

**Review by anhthu253 — COMMENTED** [review]
fine by me!

File: components/FormComponents.js (hunk 2 above)
Comment (philmetscher) [reply components/FormComponents.js:110]: no. you would fall into a loop without callback

File: pages/create-entry.js (hunk 3 above)
Comment (philmetscher) [reply pages/create-entry.js:111]: nope. Thats not what i need. the `submitReady` is for setting the button disabled or not (to be not able to submit the form)

File: components/FormComponents.js (hunk 4 above)
Comment (philmetscher) [reply components/FormComponents.js:49]: > ich glaub man muss kein event as parameter reinpacken. Der kommt automatisch mit

doch muss man (zumindest in nem callback)

**Pull request merged**